account_analytic_analysis_recurring/account_analytic_analysis_recurring.py

```diff
@@ -243,8 +243,10 @@ def recurring_create_invoice(self, cr, uid, automatic=False, context=None):
                 new_date = next_date + relativedelta(days=+interval)
             elif contract.recurring_rule_type == 'weekly':
                 new_date = next_date + relativedelta(weeks=+interval)
-            else:
+            elif contract.recurring_rule_type == 'monthly':
                 new_date = next_date + relativedelta(months=+interval)
+            else:
+                new_date = next_date+relativedelta(years=+interval)
```

Comment:
@dreispt The code is taken literally from the upstream code. It is a backport of standard Odoo in 8.0. Unfortunately Odoo SA has never cared about PEP8.